website/blog/2018-07-01-1.14.0.md

```diff
@@ -0,0 +1,758 @@
+should be especially useful for web editor applications!
+
+#### Expose `isPreviousLineEmpty` to plugins ([#4748] by [@warrenseine])
+
+Previously, we exposed `isNextLineEmpty` to plugins but not `isPreviousLineEmpty`.
+Prettier 1.14 exposes it, this can be useful for some scenarios like C# directives.
+
+### JavaScript
+
+#### Support BigInt literals ([#4697] by [@ikatyang])
+
+[BigInt](https://github.com/tc39/proposal-bigint) is now supported in the `babylon` parser.
+
+<!-- prettier-ignore -->
+```js
+const bigInt = 1n;
+```
+
+#### Support `throw` expressions ([#4695] by [@VojtechStep])
+
+[Throw expression](https://github.com/tc39/proposal-throw-expressions) is now supported in the `babylon` parser.
+
+<!-- prettier-ignore -->
+```js
+const assert = x => x || throw new Error('...');
+```
+
+#### Break JSX in arrow functions in JSX expressions ([#4601] by [@duailibe])
+
+In the prior version, JSX in arrow functions in JSX expressions followed the general break-if-fit rule,
+but it's less readable in the array iteration case.
+Prettier 1.14 force breaking them to improve the readability.
+
+<!-- prettier-ignore -->
+```jsx
+// Input
+const UsersList = ({ users }) => (
+  <section>
+    <h2>Users list</h2>
+    <ul>
+      {users.map(user => (
+        <li key={user.id}>{user.name}</li>
+      ))}
+    </ul>
+  </section>
+)
+
+// Output (Prettier 1.13)
+const UsersList = ({ users }) => (
+  <section>
+    <h2>Users list</h2>
+    <ul>{users.map(user => <li key={user.id}>{user.name}</li>)}</ul>
+  </section>
+);
+
+// Output (Prettier 1.14)
+const UsersList = ({ users }) => (
+  <section>
+    <h2>Users list</h2>
+    <ul>
+      {users.map(user => (
+        <li key={user.id}>{user.name}</li>
+      ))}
+    </ul>
+  </section>
+);
+```
+
+#### Always expand first argument if the second argument is also a call expression ([#4657] by [@duailibe])
+
+Previously, we have a special case to not break
+if there's only 2 parameters in a function and its first parameter is a function,
+this can be useful for some scenario but not if the second parameters is also a function,
+Prettier 1.14 addresses this issue.
+
+<!-- prettier-ignore -->
+```js
+// Input
+somePromise.then((args) => {
+  this.props.receiveFavoritesActions(id, [].concat(...args));
+}, ({ isCanceled }) => !isCanceled && logger.warn(`Ошибка получения действий для продукта: ${id}`));
+
+// Output (Prettier 1.13)
+somePromise.then(args => {
+  this.props.receiveFavoritesActions(id, [].concat(...args));
+}, ({ isCanceled }) => !isCanceled && logger.warn(`Ошибка получения действий для продукта: ${id}`));
+
+// Output (Prettier 1.14)
+somePromise.then(
+  args => {
+    this.props.receiveFavoritesActions(id, [].concat(...args));
+  },
+  ({ isCanceled }) =>
+    !isCanceled && logger.warn(`Ошибка получения действий для продукта: ${id}`)
+);
+```
+
+#### Handle JSX whitespace separately from fbt whitespace ([#4717] by [@karl])
+
+TODO
```

Comment:
Can someone help me with this item? I don't have too many context on it, not sure how to describe it.

Reply:
OK, I copied some sentences from the original PR to describe it, hope it should be fine.